---
title: 'Overview Datasets Project'
author: 
- Louis Coussement
date: 2020-2021
output:
  html_document:
    df_print: paged
    toc: true
  pdf_document:
    dev: png
    toc: true
header-includes: \setlength{\emergencystretch}{3em}
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.align = 'center')
knitr::opts_chunk$set(out.width = '60%')
knitr::opts_chunk$set(fig.width=12, fig.height=8)

library("ArrayExpress")
library("arrayQualityMetrics")
library("ggplot2")
library("huex10sttranscriptcluster.db")
library("limma")
library("oligo")
library("siggenes")
library("affy")
library("pd.huex.1.0.st.v2")
library("RSQLite")
library("DBI")
library("htmltools")
library("biomaRt")
library("tximport")
library("wateRmelon")
```

# Group info

Group number: 1

Group members: - Sean Barr, Halil Bora Koraaslan

```{r }
table_general_info <- rbind(c("MicroArray","Affymetrix MicroArray 1","Affymetrix MicroArray 2","ChipSeq"),
c(6,30,40,24),
c(6,30,40,10))
colnames(table_general_info) <- c("E-GEOD-30994","E-MTAB-9930","E-MTAB-9831","PRJNA311373")
rownames(table_general_info) <- c("Accesion ID","Number of Samples (original)",
"Number of Samples (analysed)")



# Link to data for Agilent MicroArray
# https://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-30994/
# Link to data for Affymetrix MicroArray (European Americans)
# https://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-71783
# Link to data for Affymetrix MicroArray (African Americans)
# https://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-71781
# Link to data for Chipseq
# https://www.ebi.ac.uk/ena/browser/view/PRJNA311373



print(table_general_info)
```

# Microarray data

## General info

cDNA microarray expression profiling between benign prostate hypertrophy and prostate cancer.


## Intensity values

Read in your microarray data and display the head and dimensions of the intensity value matrix.

```{r}
id <- "E-GEOD-30994"
sdrf <- read.delim("./microarray/E-GEOD-30994/E-GEOD-30994.sdrf.txt", check.names = FALSE, stringsAsFactors = FALSE)
orig <- getwd()
setwd("./microarray/E-GEOD-30994/")
ma_rawdata <- read.maimages(sdrf[,"Array Data File"], source="agilent", green.only = TRUE)
head(ma_rawdata)
setwd(orig)
```

## Annotation

6 samples are used in this data, 3 from benign cells and 3 from cancer cells. It will be used as a basis to see the differential expression between cancer cells and benign cells.



```{r}
(phenotypes <- sdrf[,c("Array Data File", "Characteristics [case]")])
```

# Affymetrix data 71783

## Intensity values

```{r}
id <- "E-GEOD-71783"
inputCELS <-list.celfiles("./microarray/GEOD-71783/")
data.raw <-read.celfiles(paste(rep("./microarray/GEOD-71783/",length(inputCELS)),inputCELS,sep=""))

dim(exprs(data.raw))

head(exprs(data.raw))
```

## Annotation

In this data set we have samples from the tumor and healthy tissue surrounding it. All patients had prostate cancer at the time the samples were taken. We will be investigating the differential expression between the tumor and healthy tissues.
We will also be comparing the differences between European American and African American patients to see what the differences are in the different groups (This will be compared between data set E-GEOD-71781 and E-GEOD-71783)


```{r}
sdrf <- read.delim("./microarray/GEOD-71783/E-GEOD-71783.sdrf.txt")
print(sdrf[,c("Source.Name","Comment..Sample_title.","Array.Design.REF")])
```

# Affymetrix data 71781

## Intensity values

```{r}
id <- "E-GEOD-71781"
inputCELS2 <-list.celfiles("./microarray/GEOD-71781/")
data.raw <-read.celfiles(paste(rep("./microarray/GEOD-71781/",length(inputCELS2)),inputCELS2,sep=""))

dim(exprs(data.raw))

head(exprs(data.raw))
```

## Annotation

In this data set we have samples from the tumor and healthy tissue surrounding it. All patients had prostate cancer at the time the samples were taken. We will be investigating the differential expression between the tumor and healthy tissues.
We will also be comparing the differences between European American and African American patients to see what the differences are in the different groups (This will be compared between data set E-GEOD-71781 and E-GEOD-71783)


```{r}
sdrf <- read.delim("./microarray/GEOD-71781/E-GEOD-71781.sdrf.txt")
print(sdrf[,c("Source.Name","Comment..Sample_title.","Array.Design.REF")])
```

# ChIPseq data

## General info

This ChIP-seq dataset includes H3K4me2 and H3K9me2 ChIP-seq data in 4 different conditions, siRNA for LSD1 and Luciferase, SP52509 inhibition of LSD1 and a mock treatment.



## Quality control on fastQ files


As the files were huge for my computer to run I've only done a really small portion of it here.

```{r, echo=FALSE}

htmltools::includeHTML(path="./chipseq/fastqcs/multiqc_report.html")
```

## Mapping statistics

I was met with errors while trying to map the data. Computer processor related errors, can be solved with the super computer. I'll try to finish a small part of it and send it to you if I can overcome this problem.

## Peak data

I was met with errors while trying to map the data. Computer processor related errors, can be solved with the super computer. I'll try to finish a small part of it and send it to you if I can overcome this problem.

```{r}

```

## Annotation


This data is from a paper that shows a possible treatment. It will be used to see the changes between the cancer cells and LSD1 silenced cancer cells.

```{r}
sraruntable <- read.csv("./chipseq/SraRunTable.txt")
sraruntable[,c(1,17,26)]
```
